zerver/tests/test_home.py

```diff
@@ -612,7 +612,7 @@ def test_num_queries_with_streams(self) -> None:
         self._get_home_page()
 
         # Then for the second page load, measure the number of queries.
-        with self.assert_database_query_count(49):
+        with self.assert_database_query_count(51):
```

Comment:
The increase in query count in test_home and test_event_system can be mitigated by only fetching recursive user group ids when needed within the `validate_user_access_to_subscribers_helper` function. But that would require refactoring that function to handle multiple streams and subscriptions at once, along with changing how that function is used at different places, which might be an exercise better left as a follow up.
We have optimised the code a little bit by not fetching the group ids in
case the current user is a realm admin.

This applies at two other places, please let me know if we want to fix that in this PR itself.

Reply:
Definitely not worth addressing this here. It's worth opening an issue (potentially low priority) to track a known places we can optimize this. But the goal of these query count checks is to make sure that we are careful about only using additional queries for good reasons, not to have them never increase.